pkg/query/storeset.go

```diff
@@ -81,6 +93,9 @@ var (
 		"Number of nodes with the same external labels identified by their hash. If any time-series is larger than 1, external label uniqueness is not true",
 		[]string{"external_labels"}, nil,
 	)
+	unhealthyStoreMessage = "store unhealthy or does not exists, closed gRPC client and removed from storeset"
```

Comment:
Change the first line to "removing store because it's unhealthy or does not exist". Closing gRPC client and removing from storeset is probably obvious.
Also, how can you remove store if it doesn't exist (I don't have much context on this)?

Reply:
Message updated ! For this case, from what I understand is that it compares the previous stores to the new ones returned by `getHealthyStores` and removes the ones that are not present anymore.